### PR TITLE
[DO NOT MERGE] Tmp test modifications

### PR DESCRIPTION
**Pull request by maxeeem**
@bowen-xu can you please take a look at `utils_for_test.py`. I added a function at the top that calls the reasoner directly for testing purposes. It seems to work but I wanted to get your take on it before I move further. I modified a few tests in `test_NAL1.py` just to see how it would work.

You can safely ignore all other files in this temporary PR, I will delete it afterwards.

### CODE REVIEW

**Comment by bowen-xu** [comment]
ok. I'll do it this weekend.

File: Tests/utils_for_test.py [hunk 1]
```diff
@@ -16,6 +16,29 @@
...
+    ''''''
+    # nars.reset() # TODO: uncomment when implemented
+    nars = Reasoner(100, 100)
+
+    nars.input_narsese(premise1)
+    nars.input_narsese(premise2)
+
+    tasks_all_cycles = []
+
+    for _ in range(n_cycle*10):
```
Comment (maxeeem) [inline Tests/utils_for_test.py:29]: @bowen-xu I can explain the `n_cycle*10` later ;) it is definitely not needed in the final code.

Comment (bowen-xu) [reply Tests/utils_for_test.py:29]: Sorry for the delay. There seems no problem with the function `rule_map_two_premises_`.

Comment (maxeeem) [reply Tests/utils_for_test.py:29]: No worries. So you think we should refactor all tests to use this format? (and we can come up with a better name for the function of course)

Comment (bowen-xu) [reply Tests/utils_for_test.py:29]: Yes, think so.

Comment (bowen-xu) [reply Tests/utils_for_test.py:29]: Also there are too many test files that are out of date. They should be cleaned up when getting a chance.

Comment (maxeeem) [reply Tests/utils_for_test.py:29]: Sounds good, let's discuss on Thursday. Getting tests working is important for us going forward, as we're making all these changes to the codebase trying to align it with the technical reports.

File: pynars/NARS/InferenceEngine/KanrenEngine/KanrenEngine.py [hunk 2]
```diff
@@ -412,47 +414,61 @@ def logic(self, term: Term, rule=False, substitution=False, var_intro=False, str
...
-                return Variable(VarPrefix.Independent, name[1:])
+                return create_var(name[1:], VarPrefix.Independent)
             if name[0] == '#':
-                return Variable(VarPrefix.Dependent, name[1:])
+                return create_var(name[1:], VarPrefix.Dependent)
             if name[0] == '?':
-                return Variable(VarPrefix.Query, name[1:])
+                return create_var(name[1:], VarPrefix.Query)
             else:
                 return Term(name)
```
Comment (maxeeem) [inline pynars/NARS/InferenceEngine/KanrenEngine/KanrenEngine.py:445]: @bowen-xu I modified this function as per your suggestions, seems to handle variables correctly now. Thanks!

In this line `return Term(name)`, should I be passing `do_hashing=True` or it should be left as is?

Comment (bowen-xu) [reply pynars/NARS/InferenceEngine/KanrenEngine/KanrenEngine.py:445]: My pleasure :) 
I think it's safe to left the argument to be its default value.

**Pull request closed without merging**